reconcile/test/ocm/test_utils_ocm_status_board.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any
+
+import pytest
+from pytest_mock import MockFixture
+
+from reconcile.utils.ocm.status_board import (
+    METADATA_MANAGED_BY_KEY,
+    METADATA_MANAGED_BY_VALUE,
+    create_application,
+    create_product,
+    delete_application,
+    delete_product,
+    get_managed_products,
+    get_product_applications,
+)
+
+
+@pytest.fixture
+def application_status_board() -> list[dict[str, Any]]:
+    return [
+        {
+            "id": "foo",
+            "name": "foo",
+            "fullname": "foo",
+            "metadata": {METADATA_MANAGED_BY_KEY: METADATA_MANAGED_BY_VALUE},
+        },
+        {
+            "id": "bar",
+            "foo": "bar",
+            "metadata": {METADATA_MANAGED_BY_KEY: METADATA_MANAGED_BY_VALUE},
+        },
+        {
+            "id": "oof",
+        },
+    ]
+
+
+@pytest.fixture
+def products_status_board() -> list[dict[str, Any]]:
+    return [
+        {
+            "id": "foo",
+            "name": "foo",
+            "fullname": "foo",
+            "metadata": {METADATA_MANAGED_BY_KEY: METADATA_MANAGED_BY_VALUE},
+        },
+        {
+            "id": "bar",
+            "foo": "bar",
+            "metadata": {METADATA_MANAGED_BY_KEY: METADATA_MANAGED_BY_VALUE},
+        },
+        {
+            "id": "oof",
+        },
+    ]
+
+
+def test_get_product_applications_fields(
+    mocker: MockFixture, application_status_board: list[dict[str, Any]]
+) -> None:
+    ocm = mocker.patch("reconcile.utils.ocm_base_client.OCMBaseClient")
+    mocker.patch(
+        "reconcile.utils.ocm_base_client.OCMBaseClient.get_paginated",
+        return_value=iter(application_status_board),
+    )
+    apps = get_product_applications(ocm, "foo")
+    assert len(apps) == 2
+    assert apps[0] == application_status_board[0]
+    # key foo not in APPLICATION_DESIRED_KEYS
+    assert apps[1] == {
+        "id": "bar",
+        "metadata": {METADATA_MANAGED_BY_KEY: METADATA_MANAGED_BY_VALUE},
+    }
+
+
+def test_get_managed_products(
+    mocker: MockFixture, products_status_board: list[dict[str, Any]]
+) -> None:
+    ocm = mocker.patch("reconcile.utils.ocm_base_client.OCMBaseClient")
+    mocker.patch(
+        "reconcile.utils.ocm_base_client.OCMBaseClient.get_paginated",
+        return_value=iter(products_status_board),
+    )
```

Comment:
```python
ocm = mocker.patch("reconcile.utils.ocm_base_client.OCMBaseClient", autospec=True).return_value
```
should be enough, then you can control it's method like

```python
ocm.get_patinated.return_value = iter(products_status_board)
```

Reply:
Good point! updated the tests